### PR TITLE
Fix gcc warnings

### PR DESCRIPTION
**Pull request by huangminghuang**
Try to silence most GCC 7 warnings

### CODE REVIEW

File: src/crypto/aes.cpp [hunk 1]
```diff
@@ -142,7 +142,7 @@ uint32_t aes_decoder::decode( const char* ciphertxt, uint32_t ciphertxt_len, cha
         FC_THROW_EXCEPTION( aes_exception, "error during aes 256 cbc decryption update", 
                            ("s", ERR_error_string( ERR_get_error(), nullptr) ) );
     }
-    FC_ASSERT( ciphertxt_len == plaintext_len, "", ("ciphertxt_len",ciphertxt_len)("plaintext_len",plaintext_len) );
+    FC_ASSERT( static_cast<int>(ciphertxt_len) == plaintext_len, "", ("ciphertxt_len",ciphertxt_len)("plaintext_len",plaintext_len) );
```
Comment (jgiszczak) [inline src/crypto/aes.cpp]: Prefer changing type to casting a left hand side.

File: src/crypto/elliptic_r1.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace fc { namespace crypto { namespace r1 {
           public_key_impl()
           :_key(nullptr)
           {
-          static int init = init_openssl();
+            init_openssl();
```
Comment (heifner) [inline src/crypto/elliptic_r1.cpp]: This does not look correct. The static int would make sure it is only called once.

Comment (huangminghuang) [reply src/crypto/elliptic_r1.cpp]: There's already a static in init_openssl() to ensure it only to be called once.

// in file openssl.cpp:71
int init_openssl()
{
      static openssl_scope ossl;
      return 0;
}

Comment (heifner) [reply src/crypto/elliptic_r1.cpp]: Excellent. thanks.

**Pull request merged**